flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalogBase.java

```diff
@@ -89,6 +96,34 @@ private static IMetaStoreClient getMetastoreClient(HiveConf hiveConf) {
 		}
 	}
 
+	// ------ APIs ------
+
+	/**
+	 * Validate input base table.
+	 *
+	 * @param catalogBaseTable the base table to be validated
+	 * @throws CatalogException thrown if the input base table is invalid.
+	 */
+	protected abstract void validateCatalogBaseTable(CatalogBaseTable catalogBaseTable)
+		throws CatalogException;
+
+	/**
+	 * Create a CatalogBaseTable from a Hive table.
+	 *
+	 * @param hiveTable a Hive table
+	 * @return a CatalogBaseTable
+	 */
+	protected abstract CatalogBaseTable createCatalogBaseTable(Table hiveTable);
```

Comment:
I think we can just name the method createCatalogTable() which sounds more general, and as a direct counterpart of createHiveTable(). However, this is very minor though.

Reply:
This method will return either a CatalogTable or CatalogView, that's why it's named "createCatalogBaseTable() " and "createCatalogTable()" may be a bit misleading